huaweicloud/services/oms/resource_huaweicloud_oms_migration_sync_task.go

```diff
@@ -0,0 +1,413 @@
+package oms
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+
+	omsmodel "github.com/huaweicloud/huaweicloud-sdk-go-v3/services/oms/v2/model"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceMigrationSyncTask() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMigrationSyncTaskCreate,
+		ReadContext:   resourceMigrationSyncTaskRead,
+		UpdateContext: resourceMigrationSyncTaskUpdate,
+		DeleteContext: resourceMigrationSyncTaskDelete,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"src_region": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"src_bucket": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"src_ak": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"src_sk": {
+				Type:      schema.TypeString,
+				Sensitive: true,
+				Required:  true,
+				ForceNew:  true,
+			},
+
+			"dst_bucket": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"dst_ak": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"dst_sk": {
+				Type:      schema.TypeString,
+				Sensitive: true,
+				Required:  true,
+				ForceNew:  true,
+			},
+			"src_cloud_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
```

Comment:
it's better to delete `ValidateFunc` and `Default` because the API maybe updated later.